mdtraj/core/topology.py

```diff
@@ -1905,8 +1905,13 @@ def float_to_bond_type(bond_float: float) -> Singleton | None:
     """
     all_bond_types = [Single, Double, Triple, Aromatic, Amide]
     for bond_type in all_bond_types:
-        if float(cast(Singleton, bond_type)) == float(bond_float):  # explicitly cast to Singleton to avoid mypy error
-            return cast(Singleton, bond_type)
+        try:
+            if float(cast(Singleton, bond_type)) == float(bond_float):
+                # explicitly cast to Singleton to avoid mypy error
+                return cast(Singleton, bond_type)
+        except TypeError:
```

Comment:
Are we just trying to catch `bond_float=None` here? If so, it may be faster/cleaner to check for this upfront and return `None` instead of going through the loop.

Besides this, this function seems like it would be cleaner by using a dictionary mapping rather than the current loop.

```python
def float_to_bond_type(bond_float: float | None) -> Singleton | None:
    
bond_map: dict[float, Singleton] = {
        float(Single): Single,
        float(Double): Double,
        float(Triple): Triple,
        float(Aromatic): Aromatic,
        float(Amide): Amide,
    }
    try:
        return bond_map.get(float(bond_float), None)
    except (TypeError, ValueError):
        return None
```